src/tests/test_config.py

```diff
@@ -97,6 +101,29 @@ def find_subclasses(obj):
 
             assert vars(first_conf.optimizations) == vars(second_conf.optimizations)
 
+    def test_optimized_instructions_metric(self):
+        conf = CompilerConfig()
+        conf.metrics = conf.metrics | conf.metrics.OptimizedInstructions
```

Comment:
would explicitly set it to `conf.metrics.OptimizedInstructions`, no chance for default metrics to interfere if they ever shift.

Reply:
also default metrics are tested separately in the following test

Reply:
Metrics will never interfere with each other, it doesn't test for them, and doing a flag append is the natural way you would enable the feature so testing that is meaningful. Not a bad suggestion though!

Reply:
ah yes, my mind went straight to Python `or` ^^ the flag append operator makes sense :smiley: